docs/features/exploiting_sparsity.ipynb

```diff
@@ -0,0 +1,246 @@
+   "source": [
+    "# Exploiting Sparsity for Faster Derivative Calculation\n",
+    "\n",
+    "A key feature of collocation algorithms such as high-order Gauss-Lobatto collocation or the Radau pseudospectral method is that they exhibit a large degree of *sparsity* in the total Jacobian.\n",
+    "By modeling the state-time histories as a series of polynomial segments, the collocation defect constraints within each segment are largely dependent only on the state and control values within the same segment.\n",
+    "Pseudospectral optimization tools such as SOCS, OTIS, and GPOPS-II have long used the notion of *sparse finite differences* to perturb multiple independent variables simultaneously when approximating the constraint Jacobian.\n",
+    "This can significantly reduce the computational effort required to approximate the entire Jacobian via finite difference.\n",
+    "\n",
+    "The unique way in which OpenMDAO assembles analytic derivatives across the problem allows us to use a similar approach to provide the analytic constraint Jacobian.\n",
+    "This approach can reduce the time required to solve moderately-sized optimal control problems by orders of magnitude and make the convergence more robust.\n",
+    "\n",
+    "OpenMDAO uses a [\"coloring\" algorithm](https://openmdao.org/newdocs/versions/latest/features/core_features/working_with_derivatives/simul_derivs.html) to determine which variables can be perturbed simultaneously to determine the total constraint Jacobian.\n",
+    "Variables in the same \"color\" each impact a unique constraint, such that when all the variables are perturbed we can be assured that any change in the constraint vector is due to at most one scalar variable.\n",
+    "Since OpenMDAO uses a linear solver to assemble to total derivative Jacobian, coloring reduces the number of linear solves from one per variable (or constraint in adjoint mode) to one per *color*.\n",
+    "In the brachistochrone example below this reduces the number of linear solves from 1001 to 9.\n",
+    "This capability makes problems of moderate size run orders of magnitude faster, and can make intractably large problems tractable.\n",
+    "\n",
+    "```{Note}\n",
+    "While some optimizers (SNOPT and IPOPT) are particularly adept at dealing with large, sparse nonlinear programming problems, coloring can still benefit drivers which do not account for sparsity (such as SLSQP) since it significantly reduces the cost of computing the Jacobian.\n",
+    "```\n",
+    "\n",
+    "## Step 1: Using OpenMDAO's Coloring Capability\n",
+    "\n",
+    "OpenMDAO supports dynamic coloring, meaning it can automatically run the Jacobian coloring algorithm before handing the problem to the optimizer.\n",
+    "To enable this capability, simply add the following line to the driver."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "driver.declare_coloring()"
```

Comment:
I get an exception in the results from this cell. This shows no output in the online docs right now, so I am guessing this isn't intended to be an executable code cell anyway.

Reply:
Fixed